src/components/PassengerShowcaseForm.js

```diff
@@ -56,11 +54,92 @@ const PassengerShowcaseForm = () => {
           videoNumber
         }
       }
+      tracks: allTrack {
+        edges {
+          node {
+            title
+            slug
+            chapters {
+              title
+              videos {
+                title
+                slug
+                canContribute
+                canonicalTrack {
+                  slug
+                }
+              }
+            }
+            videos {
+              title
+              slug
+              canContribute
+              canonicalTrack {
+                slug
+              }
+            }
+          }
+        }
+      }
     }
   `);
 
+  const tracks = data.tracks.edges
+    .map((edge) => edge.node)
+    .map((track) => {
+      // gather all videos from chapters (main track)
+      if (!track.videos && track.chapters) {
+        track.videos = track.chapters.reduce((acc, chapter) => {
+          return acc.concat(chapter.videos);
+        }, []);
+      }
+      delete track.chapters;
+      return track;
+    })
+    .map((track) => {
+      // filter out videos that can't be contributed to
+      track.videos = track.videos.filter((video) => video.canContribute);
+      // keep only videos that belong to this track
+      track.videos = track.videos.filter(
```

Comment:
Maybe you could chain the two filters and assign to `track.videos` only once? I initially thought it was a mistake and the second filter was undoing the first. Not a big deal.